### PR TITLE
scrollbound animations and AMP-host for position-observer

### LINKED ISSUE #8737: scroll-bound amp-animation (part 1)
Broken down from: https://github.com/ampproject/amphtml/issues/8411

Limit (for now) to:
a4a case
non-configurable visibility

### PR DESCRIPTION
**Pull request by aghassemi**
Closes #8737 

sb2

### CODE REVIEW

File: src/service/position-observer-impl.js [hunk 1]
```diff
@@ -0,0 +1,402 @@
+import {AmpContext} from '../../3p/ampcontext';
```
Comment (aghassemi) [inline src/service/position-observer-impl.js]: Please don't review this file in this PR. The changes in this file are to be merged with #8940 and this PR will be dependent on #8940

**Comment by aghassemi** [comment]
/to @dvoytenko (this requires #8940 to be merged first)

File: extensions/amp-animation/0.1/amp-animation.js [hunk 2]
```diff
@@ -282,8 +285,37 @@ export class AmpAnimation extends AMP.BaseElement {
       this.finish();
     }
   }
-}
 
+  /**
+   * @private
+   */
+  setupScrollboundAnimatins_() {
```
Comment (dvoytenko) [inline extensions/amp-animation/0.1/amp-animation.js]: nit: spelling.

File: extensions/amp-animation/0.1/scrollbound-player.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+export class ScrollboundPlayer {
+
+  constructor(request) {
```
Comment (dvoytenko) [inline extensions/amp-animation/0.1/scrollbound-player.js]: jsdoc

File: extensions/amp-animation/0.1/scrollbound-scene.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+import {
+  installPositionObserverServiceForDoc,
+  PositionObserverFidelity,
+} from '../../../src/service/position-observer-impl';
+import {getServiceForDoc} from '../../../src/service';
+
+export class ScrollboundScene {
+
+  constructor(ampdoc, element, onScroll, onDurationChanged) {
```
Comment (dvoytenko) [inline extensions/amp-animation/0.1/scrollbound-scene.js]: jsdoc

File: extensions/amp-animation/0.1/web-animations.js [hunk 5]
```diff
@@ -175,6 +190,44 @@ export class WebAnimationRunner {
   }
 
   /**
+   */
+  scrollTick(pos) {
```
Comment (dvoytenko) [inline extensions/amp-animation/0.1/web-animations.js]: Ok for now. But I'm leaning toward all sub-animations should be of the same ticker as a rule and different tickers is more of an exception (i.e. maybe a separate animation). Topic for future discussion.

**Pull request merged**